pgxn/neon/pagestore_smgr.c

```diff
@@ -1903,7 +1903,6 @@ neon_wallog_pagev(SMgrRelation reln, ForkNumber forknum, BlockNumber blocknum,
 		log_pages = true;
 	}
 	else if (XLogInsertAllowed() &&
-			 !ShutdownRequestPending &&
 			 (forknum == FSM_FORKNUM || forknum == VISIBILITYMAP_FORKNUM))
```

Comment:
I wonder if there was some other reason for the `!ShutdownRequestPending` condition, other than just making shutdown go faster? I can't think of any. It goes all the way back to [this commit](https://github.com/neondatabase/postgres/commit/55ebeac1f48c7e8cb64df373fb28e6a0c50bdafe), which based on the commit message and the other changes in it seem unrelated. Do you happen to remember anything about that?

Reply:
Obviously the main reason was to avoid panic in checkpoint:
```
PG:2025-04-04 10:27:38.580 GMT [55738] LOG:  shutting down
PG:2025-04-04 10:27:38.583 GMT [55738] LOG:  checkpoint starting: shutdown immediate
PG:2025-04-04 10:27:38.585 GMT [55738] PANIC:  concurrent write-ahead log activity while database system is shutting down
PG:2025-04-04 10:27:38.585 GMT [55737] LOG:  checkpointer process (PID 55738) was terminated by signal 6: Abort trap: 6
```

This is what I am preventing now by changes in Postgres.

Reply:
I guess :-D. It seems unrelated to the rest of the commit though. Maybe you hit that panic when testing the other changes and included it in the same commit because of that.